### PR TITLE
implement "svc.sh" script to install, start, stop, uninstall linux se…

### PR DESCRIPTION
**Pull request by stiliev**
implement "svc.sh" script to install, start, stop, uninstall linux service, similar to OSX service

### CODE REVIEW

File: src/Agent.Listener/Configuration/SystemdControlManager.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                File.WriteAllText(svcShPath, svcShContent, Encoding.UTF8);
+
+                var unixUtil = HostContext.CreateService<IUnixUtil>();
+                unixUtil.Chmod("755", svcShPath).GetAwaiter().GetResult();
+
+                SvcSh("install");
+                _term.WriteLine(StringUtil.Loc("ServiceConfigured", settings.ServiceName));
+            }
+            catch (UnauthorizedAccessException ex)
+            {
```
Comment (stiliev) [inline src/Agent.Listener/Configuration/SystemdControlManager.cs]: this exception is no longer thrown (it worked with pinvoke commands before). i need to find out how to handle permission errors

File: src/Agent.Listener/Configuration/ConfigurationManager.cs [hunk 2]
```diff
@@ -247,6 +247,31 @@ public AgentSettings LoadSettings()
...
+                                                             { IOUtil.GetConfigFilePath(), "770"},
+                                                             { IOUtil.GetCredFilePath(), "770" },
+                                                         };
+                    var unixUtil = HostContext.CreateService<IUnixUtil>();
+                    foreach (var file in filesToChange)
+                    {
+                        await unixUtil.Chown(uidValue, gidValue, file.Key);
+                        await unixUtil.Chmod(file.Value, file.Key);
+                    }
+                }
```
Comment (bryanmacfarlane) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: And if they are Null Or Empty?  Error?

Comment (stiliev) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: If uid or gid is empty, it means we run without sudo. This should be fine if we don't run as a service.

**Comment by stiliev** [comment]
Moved service related settings in a ".Service" file. Need more testing on OSX and Windows.

File: src/Misc/layoutbin/darwin.svc.sh.template [hunk 3]
```diff
@@ -40,6 +42,14 @@ function install()
     cp "${TEMP_PATH}" "${PLIST_PATH}" || "failed to copy plist"
     cp ./bin/runsvc.sh ./runsvc.sh
     chmod 755 ./runsvc.sh
+
+    echo '
+    {
+        "RunAsService": true,
```
Comment (bryanmacfarlane) [inline src/Misc/layoutbin/darwin.svc.sh.template]: Why is this json?  Of course RunAsService is true, this file exists right?

File: src/Misc/layoutbin/darwin.svc.sh.template [hunk 4]
```diff
@@ -40,6 +42,14 @@ function install()
...
+        "ServiceName": "'${SVC_NAME}'",
+        "ServiceDisplayName": "'${SVC_DESCRIPTION}'"
```
Comment (bryanmacfarlane) [inline src/Misc/layoutbin/darwin.svc.sh.template]: Why?  Service display name is useless - that's in the unit file.

File: src/Misc/layoutbin/darwin.svc.sh.template [hunk 5]
```diff
@@ -40,6 +42,14 @@ function install()
     cp "${TEMP_PATH}" "${PLIST_PATH}" || "failed to copy plist"
     cp ./bin/runsvc.sh ./runsvc.sh
     chmod 755 ./runsvc.sh
+
+    echo '
+    {
+        "RunAsService": true,
+        "ServiceName": "'${SVC_NAME}'",
```
Comment (bryanmacfarlane) [inline src/Misc/layoutbin/darwin.svc.sh.template]: OK.  This is the only piece of data that's interesting.  This is a shell script and this file is only used as (1) it's existance implies it's a service and (2) info for user - what's the svc name.  As we discussed, this file only needs to contain a string, the name of the svc.

File: src/Misc/layoutbin/systemd.svc.sh.template [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	# Since we started with sudo, runsvc.sh will be owned by root. Change this to current login user.    
+    cp ./bin/runsvc.sh ./runsvc.sh
+    chown ${SUDO_UID:-$UID}:${SUDO_GID} ./runsvc.sh
+    chmod 755 ./runsvc.sh
+
+    systemctl enable ${SVC_NAME} || "failed to enable ${SVC_NAME}"
+
+    echo '
+    {
+        "RunAsService": true,
```
Comment (bryanmacfarlane) [inline src/Misc/layoutbin/systemd.svc.sh.template]: Ditto

**Comment by bryanmacfarlane** [comment]
I'm going to merge this so we can do coding and testing in parallel

File: src/Misc/layoutbin/darwin.svc.sh.template (hunk 3 above)
Comment (stiliev) [inline src/Misc/layoutbin/darwin.svc.sh.template]: I moved the properties from .Agent to .Service, but indeed the only important thing is if the file exists. I agree with your comments below.

File: src/Misc/layoutbin/darwin.svc.sh.template (hunk 5 above)
Comment (stiliev) [inline src/Misc/layoutbin/darwin.svc.sh.template]: I forgot we discussed that the file needs only a string. This makes sense and I see you made the change already.

**Pull request merged**